### PR TITLE
resource_control: fetch cpu quota metrics from store instead of prometheus

### LINKED ISSUE #49174: static calibrate resource should not depend on prometheus data
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
deploy a tidb cluster with tiup playground `--without-minitor` and run:
```
mysql> calibrate resource;
ERROR 1105 (HY000): There is no CPU quota metrics, query metric error: [domain:9009]Prometheus address is not set in PD and etcd
```

### 4. What is your TiDB version? (Required)

nightly

### PR DESCRIPTION
**Pull request by glorv**
### What problem does this PR solve?

Issue Number: close #49174

Problem Summary:

### What changed and how does it work?
Use the /metrics api to fetch realtime metrics from tikv and tiflash and parse cpu quota from the metrics. This can avoid depending on prometheus data for static `calibrate resource`.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the bug that static `calibrate resource` depends on prometheus data.
```

### CODE REVIEW

**Comment by glorv** [comment]
@nolouch @CabinfeverB @HuSharp PTAL

File: pkg/executor/internal/calibrateresource/calibrate_resource.go [hunk 1]
```diff
@@ -81,12 +89,21 @@ var (
 	}
 )
 
+const (
+	// serverTypeTiDB is tidb's instance type name
+	serverTypeTiDB = "tidb"
+	// serverTypeTiDB is tikv's instance type name
+	serverTypeTiKV = "tikv"
+	// serverTypeTiDB is tiflash's instance type name
+	serverTypeTiFlash = "tiflash"
```
Comment (HuSharp) [inline pkg/executor/internal/calibrateresource/calibrate_resource.go:98]:
```suggestion
	// serverTypeTiDB is tidb's instance type name
	serverTypeTiDB = "tidb"
	// serverTypeTiKV is tikv's instance type name
	serverTypeTiKV = "tikv"
	// serverTypeTiFlash is tiflash's instance type name
	serverTypeTiFlash = "tiflash"
```

File: pkg/executor/internal/calibrateresource/calibrate_resource.go [hunk 2]
```diff
@@ -81,12 +89,21 @@ var (
...
+)
+
 // the resource cost rate of a specified workload per 1 tikv cpu.
 type baseResourceCost struct {
 	// represents the average ratio of TiDB CPU time to TiKV CPU time, this is used to calculate whether tikv cpu
 	// or tidb cpu is the performance bottle neck.
 	tidbToKVCPURatio float64
-	// the kv CPU time for calculate RU, it's smaller than the actual cpu usage. The unit is seconds.
+	// the kv CPU time for calcula[te R]U, it's smaller than the actual cpu usage. The unit is seconds.
```
Comment (HuSharp) [inline pkg/executor/internal/calibrateresource/calibrate_resource.go:106]:
```suggestion
	// the kv CPU time for calculate RU, it's smaller than the actual cpu usage. The unit is seconds.
```

**Review by CabinfeverB — APPROVED** [review]

**Comment by glorv** [comment]
@nolouch @HuSharp PTAL

**Review by HuSharp — APPROVED** [review]
LGTM!

**Review by nolouch — APPROVED** [review]
lgtm

**Pull request merged**